benchmarks/src/jmh/java/io/grpc/benchmarks/netty/AbstractBenchmark.java

```diff
@@ -90,6 +95,58 @@ public int bytes() {
     NIO, LOCAL;
   }
 
+  private static final InetAddress BENCHMARK_ADDR;
+
+  /**
+   * Resolve the address bound to the benchmark interface. Currently we assume it's a
+   * child interface of the loopback interface with the term 'benchmark' in its name.
+   *
+   * This allows traffic shaping to be applied to an IP address and to have the benchmarks
+   * detect it's presence and use it. E.g for Linux we can apply netem to a specific IP to
+   * do traffic shaping, bind that IP to the loopback adapter and then apply a label to that
+   * binding to it appears as a child interface.
+   *
+   * sudo tc qdisc del dev lo root
+   * sudo tc qdisc add dev lo root handle 1: prio
+   * sudo tc qdisc add dev lo parent 1:1 handle 2: netem delay 0.1ms rate 10gbit
+   * sudo tc filter add dev lo parent 1:0 protocol ip prio 1 u32 match ip dst 127.127.127.127 flowid 2:1
+   * sudo ip addr add dev lo 127.127.127.127/32 label lo:benchmark
+   *
+   */
+  static {
+    InetAddress tmp = null;
+    try {
+      Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+      while (networkInterfaces.hasMoreElements()) {
+        NetworkInterface networkInterface = networkInterfaces.nextElement();
+        if (networkInterface.isLoopback()) {
+          Enumeration<NetworkInterface> subInterfaces = networkInterface.getSubInterfaces();
+          while (subInterfaces.hasMoreElements()) {
+            NetworkInterface subLoopback = subInterfaces.nextElement();
+            if (subLoopback.getDisplayName().contains("benchmark")) {
+              tmp = subLoopback.getInetAddresses().nextElement();
+              System.out.println("\n\nSUCCESS: Resolved benchmark address to " + tmp + " on " +
```

Comment:
Let's not say SUCCESS, since it seems like the benchmark completed? Simply the notice "Resolved benchmark..." seems sufficient.